llpc/util/llpcError.cpp

```diff
@@ -117,6 +108,26 @@ std::error_code resultToErrorCode(Result result) {
   return {static_cast<int>(result), GlobalCategoryId};
 }
 
+// =====================================================================================================================
+// Prints the error message in `err` to LLPC_ERRS and consumes the error.
+//
+// @param err : The error to handle. This must not be an `ErrorSuccess`.
+// @returns: The underlying `Result` when `err` is a `ResultError`, `Result::ErrorUnknown` otherwise.
+Result reportError(Error &&err) {
+  // For details on llvm error handling, see https://llvm.org/docs/ProgrammersManual.html#recoverable-errors.
+  assert((bool)err && "llvm::ErrorSuccess is not an error");
```

Comment:
Would a `static_cast<bool>(err)` be more C++-like?

Reply:
Yes, but no cast are actually needed at all -- assert does a double negation internally. I removed the cast.